src/backendCommonsLib/src/FlowSecret.js

```diff
@@ -0,0 +1,49 @@
+const _ = require('lodash');
+const { URL } = require('url');
+
+const toBase64 = str => Buffer.from(str).toString('base64');
+const fromBase64 = str => Buffer.from(str, 'base64').toString();
+
+class FlowSecret {
+    constructor({ metadata = {}, data = {} } = {}) {
+        Object.assign(this, {
+            metadata,
+            data
+        });
+    }
+
+    get id() {
+        return _.get(this, 'metadata.name');
+    }
+
+    get amqpUsername() {
+        const { username } = new URL(_.get(this, 'data.AMQP_URI'));
+        return username;
+    }
+
+    toDescriptor() {
+        return {
+            apiVersion: 'v1',
+            kind: 'Secret',
+            metadata: this.metadata,
```

Comment:
Could you please use ownership mechanism like in job's metadata.
This gives us possibility to interlink secrets with flows as it's used in k8s